modAionImpl/src/org/aion/zero/impl/AionBlockchainImpl.java

```diff
@@ -658,8 +685,10 @@ AionBlock createNewBlockInternal(AionBlock parent, List<AionTransaction> txs, bo
          * Calculate the gas used for the included transactions
          */
         long totalEnergyUsed = 0;
+        BigInteger totalTransactionFee = BigInteger.ZERO;
         for (AionTxExecSummary summary : preBlock.summaries) {
             totalEnergyUsed = totalEnergyUsed + summary.getNrgUsed().longValueExact();
+            totalTransactionFee = totalTransactionFee.add(summary.getNrgUsed().multiply(summary.getFee()));
```

Comment:
This calculates the totalTransactionFee incorrectly due to the implementation of getFee().

getFee() already multiplies NrgUsed * NrgPrice, the additional getNrgUsed multiplication causes an incorrect total transaction fee. This can simply be changed to summary.getFee().